File: LeetCode Solutions/April 2022 LeetCoding Challenge/Convert BST to Greater Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int total = 0;
    TreeNode* convertBST(TreeNode* root) {
        if(root == NULL){
            return NULL;
        }
        if(root->right != NULL){
            convertBST(root->right);
        }
        total += root->val;
        root->val = total;
        if(root->left != NULL){
            convertBST(root->left);
        }
        return root;
    }
};

File: LeetCode Solutions/LeetCoding Challenge/February 2021 LeetCoding Challenge/Convert BST to Greater Tree/Convert BST to Greater Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int total = 0;
    public TreeNode convertBST(TreeNode root) {
        if (root == null){
            return null;
        }
        if (root.right != null){
            convertBST(root.right);
        }
        total += root.val;
        root.val = total;
        if (root.left != null){
            convertBST(root.left);
        }
        return root;
    }
}